packages/js-client/src/context.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  SimpleStorageContextParams,
+  SimpleStorageContextState,
+  SimpleStorageOverridenState,
+} from "./types";
+import { Context, ContextCore } from "@aragon/sdk-client-common";
+
+// set your defaults here or import them from a package
+const DEFAULT_SIMPLE_STORAGE_PLUGIN_ADDRESS =
+  "0x1234567890123456789012345678901234567890";
+const DEFAULT_SIMPLE_STORAGE_Repo_ADDRESS =
+  "0x2345678901234567890123456789012345678901";
+
+export class SimpleStorageContext extends ContextCore {
+  // super is called before the properties are initialized
+  // so we initialize them to the value of the parent class
+  protected state: SimpleStorageContextState = this.state;
+  protected overriden: SimpleStorageOverridenState = this.overriden;
+  constructor(
+    contextParams?: Partial<SimpleStorageContextParams>,
+    aragonContext?: Context,
+  ) {
+    // parent contructor will call this.set(contextParams)
+    // so we don't need to call it again
+    super(contextParams);
+    if (aragonContext) {
+      // copy the context properties to this
+      Object.assign(this, aragonContext);
```

Comment:
Wouldn't this replace `this.state` and `this.overridden` completely and wipe the data set on the `super(contextParams)` line?

Reply:
Yep, that why i call it again inside the set function, a bit complex but I wanted to give priority to the context params over the context and TS does not allow you to have a contructor without calling super

Reply:
This might be confusing for people approaching the template. Wouldn't a `@ts-ignore` silence this constructor+super warning, so that we did things in the right order?

Or maybe running `super({})` with a comment explaining that we do this to silence the warning, but not shifting the attention away from the right way of using the context

Reply:
a ts-ignoer wont solve anything because then it will complaain because you wouldn have access to `this`, we can leeave a `super()`

Reply:
Up to you :+1:

Reply:
The other part the comment was about the shallow copy being done here

If the values of aragonContext change, this class will be affected as well which is a side effect. Isn't it possible to do a destructured copy similar to `Object.assign(this, {...aragonContext})`? Or even just getting the `aragonContext` values and doing a manual `super.set(...)`?